From 5dbd1d56b3e8beb2eaf9f13da60e0c6a35717c0f Mon Sep 17 00:00:00 2001
From: Binh Nguyen <binh.nguyen.jz@renesas.com>
Date: Wed, 10 Mar 2021 21:41:09 +0700
Subject: [PATCH 094/135] arm64: dts: r9a07g044l: add fcpvd0, vspd0, display
 nodes

This commit adds support fcpvd0, vspd0 and display unit nodes for
r9a07g044l SoC.

Signed-off-by: Hien Huynh <hien.huynh.px@renesas.com>
---
 arch/arm64/boot/dts/renesas/r9a07g044l.dtsi | 48 +++++++++++++++++++++++++++++
 1 file changed, 48 insertions(+)

diff --git a/arch/arm64/boot/dts/renesas/r9a07g044l.dtsi b/arch/arm64/boot/dts/renesas/r9a07g044l.dtsi
index ef58822..525c3fb 100644
--- a/arch/arm64/boot/dts/renesas/r9a07g044l.dtsi
+++ b/arch/arm64/boot/dts/renesas/r9a07g044l.dtsi
@@ -422,6 +422,54 @@
 			resets =  <&cpg R9A07G044L_CLK_GPT>;
 			status = "disabled";
 		};
+
+		vspd0: vsp@10870000 {
+			compatible = "renesas,vsp2";
+			reg = <0 0x10870000 0 0x10000>;
+			interrupts = <GIC_SPI 149 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&cpg CPG_MOD R9A07G044L_CLK_LCDC>;
+			resets = <&cpg R9A07G044L_CLK_LCDC>;
+			power-domains = <&cpg>;
+			renesas,fcp = <&fcpvd0>;
+		};
+
+		fcpvd0: fcp@10880000 {
+			compatible = "renesas,fcpv";
+			reg = <0 0x10880000 0 0x10000>;
+			clocks = <&cpg CPG_MOD R9A07G044L_CLK_LCDC>;
+			resets = <&cpg R9A07G044L_CLK_LCDC>;
+			power-domains = <&cpg>;
+		};
+
+		du: display@feb00000 {
+			compatible = "renesas,du-r9a04g077l";
+			reg = <0 0x10890000 0 0x10000>;
+			interrupts = <GIC_SPI 152 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&cpg CPG_MOD R9A07G044L_CLK_LCDC>;
+			clock-names = "du.0";
+			resets = <&cpg R9A07G044L_CLK_LCDC>;
+			reset-names = "du.0";
+			vsps = <&vspd0 0>;
+			power-domains = <&cpg>;
+			status = "disabled";
+
+			ports {
+				#address-cells = <1>;
+				#size-cells = <0>;
+
+				port@0 {
+					reg = <0>;
+					du_out_rgb: endpoint {
+					};
+				};
+
+				port@1 {
+					reg = <1>;
+					du_out_dsi0: endpoint {
+					};
+				};
+			};
+		};
 	};
 
 	thermal-zones {
-- 
2.7.4

